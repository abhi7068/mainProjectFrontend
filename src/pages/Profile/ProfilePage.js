import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';
import Tab from '@material-ui/core/Tab';
import Tabs from '@material-ui/core/Tabs';
import Typography from '@material-ui/core/Typography';
import AccountCircle from '@material-ui/icons/AccountCircle';
// import Box from '@material-ui/core/Box';
import EmailIcon from '@material-ui/icons/Email';
import './profile.css';
import SaveIcon from '@material-ui/icons/Save';
import { withStyles } from '@material-ui/styles';
import React, { Component } from 'react';
import TabPanel from '../../atoms/TabPanel';
import {
  getUserById,
  UpdateById,
  updateProfilePic,
} from '../../services/UserService';
import PropTypes from 'prop-types';
import { TextField } from '@material-ui/core';
import Avatar from '@material-ui/core/Avatar';
import ImageUpload from '../../molecules/ImageUpload';
import EditIcon from '@material-ui/icons/Edit';
import IconButton from '@material-ui/core/IconButton';
import DialogBox from '../../organisms/DialogBox';
const useStyles = {
  container: {
    flex: 1,
    backgroundColor: '',
    alignItems: 'center',
    justifyContent: 'center',
  },

  font_Design: {
    color: 'text.primary',
  },
  tab_design: {
    color: 'black',
    height: 20,
    padding: '0 30px',
    margin: '0 20px',
  },
  show: {
    cursor: 'initial',
  },
  root: {
    flexGrow: 1,
    display: 'flex',
    height: 500,
    tecolor: 'white',
    align: 'center',
    padding: '50px 40px',
    // borderTop: '20px solid #313a43',
    // borderBottom: '20px solid #313a43',
    margin: 2,
  },
  text: {
    minWidth: 150,
    display: 'flex',
    border: '0 0 0 0',
    color: 'black',
    padding: 100,
  },
  bullet: {
    display: 'inline-block',
    margin: '0 2px',
    transform: 'scale(0.8)',
  },
  title: {
    fontSize: 14,
  },
  pos: {
    marginBottom: 12,
  },
};
class Profie extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: 0,
      openDialog: false,
      data: {},
      hidden: true,
    };
    this.onModelChange = this.onModelChange.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.sendDetails = this.sendDetails.bind(this);
    this.showEdit = this.showEdit.bind(this);
    this.closeDialogBox = this.closeDialogBox.bind(this);
    this.openDialogBox = this.openDialogBox.bind(this);
    this.updatePoints = this.updatePoints.bind(this);
  }

  closeDialogBox() {
    this.setState((prevState) => ({ ...prevState, openDialog: false }));
  }

  openDialogBox() {
    this.setState((prevState) => ({ ...prevState, openDialog: true }));
  }

  updatePoints(newPoints) {
    this.setState((prevState) => {
      return {
        ...prevState,
        data: {
          ...prevState.data,
          yoyobalance: +prevState.data.yoyobalance + +newPoints,
        },
      };
    });
    const getResponseDetails = localStorage.getItem('storeResponseDetails');
    const getResponseDetailsAfterParse = JSON.parse(getResponseDetails);
    UpdateById(getResponseDetailsAfterParse.id, this.state.data);
    setTimeout(() => this.closeDialogBox(), 1000);
  }

  sendDetails() {
    const getResponseDetails = localStorage.getItem('storeResponseDetails');
    const getResponseDetailsAfterParse = JSON.parse(getResponseDetails);
    UpdateById(getResponseDetailsAfterParse.id, this.state.data);
  }

  showEdit() {
    const { hidden } = this.state;
    this.setState({
      hidden: !hidden,
    });
  }

  handleChange(event, newValue) {
    event.preventDefault();
    this.setState({
      value: newValue,
    });
  }

  onModelChange(event) {
    const name = event.target.name;
    const value = event.target.value;
    this.setState({ data: { ...this.state.data, [name]: value } });
  }

  componentDidMount() {
    const getResponseDetails = localStorage.getItem('storeResponseDetails');

    if (getResponseDetails === null) {
    } else {
      const getResponseDetailsAfterParse = JSON.parse(getResponseDetails);

      if (
        getResponseDetailsAfterParse.isAdmin === false &&
        getResponseDetailsAfterParse.status === true
      ) {
        getUserById(getResponseDetailsAfterParse.id).then((res) => {
          this.setState({
            data: res.data,
          });
        });
      }
    }
  }

  render() {
    const { value, data } = this.state;
    //
    const { showEdit } = this;
    const { handleChange, sendDetails, onModelChange } = this;
    const { classes } = this.props;
    return (
      <div className="root" style={useStyles.root}>
        <Tabs
          className="tab"
          orientation="vertical"
          variant="scrollable"
          value={value}
          onChange={handleChange}
          aria-label="Vertical tabs"
        >
          <Tab
            className={classes.tab_design}
            label={
              <Typography className="a">
                {' '}
                Welcome <br />
                <hr />
                <hr />
                {data.name}
              </Typography>
            }
          />
          <Tab
            className={classes.tab_design}
            label={<a className="a">Edit Your profile</a>}
          />
          <Tab
            label={<a className="a">yoyo balance</a>}
            className={classes.tab_design}
          />
          <Tab
            className={classes.tab_design}
            label={<a className="a">sent order</a>}
          />
          <Tab
            className={classes.tab_design}
            label={<a className="a">recieved history</a>}
          />
        </Tabs>
        <TabPanel value={value} index={0}>
          <Grid container spacing={3} alignItems="flex-end">
            <Grid item style={{ float: 'right' }}>
              <Avatar
                alt="Cindy Baker"
                type=""
                // src={data.profilePictrue}
                src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcSkkolxSNKrb80PEcRLo2ocTotIoM1TmxA_U02TJrM0UGSdwP9H&usqp=CAU"
                style={{ width: '90px', height: '100px', float: 'right' }}
                onChange={onModelChange}
                placeholder="Cindy Baker"
              />
            </Grid>
          </Grid>
          <Typography
            className={classes.title}
            color="textSecondary"
            gutterBottom
          >
            Gift the products as a gift card! When you purchase, there wil
          </Typography>
          <Typography className={classes.pos} color="textSecondary">
            be a form to indicate who the gift is for, and you can send gift
            cards to
          </Typography>
          <Typography className={classes.pos} color="textSecondary">
            your friend. Buying for a member or not a member, do not worry we
            will send them{' '}
            <span style={{ color: 'skyblue' }}>registration link </span>
            also.
          </Typography>
          <Typography variant="body2" component="p">
            <br />
          </Typography>
        </TabPanel>
        <TabPanel value={value} index={1}>
          <div>
            <Grid container spacing={3} alignItems="flex-end">
              <Grid item style={{ float: 'right' }}>
                <Avatar
                  alt="Cindy Baker"
                  type=""
                  src={data.profilePictrue}
                  /* src="https://thumbs.dreamstime.com/z/default-avatar-placeholder-profile-icon-male-eps-file-easy-to-edit-default-avatar-placeholder-profile-icon-male-139556753.jpg" */
                  style={{ width: '90px', height: '100px', float: 'right' }}
                  onChange={onModelChange}
                  placeholder="Cindy Baker"
                />
              </Grid>
              <Grid>
                {this.state.hidden ? (
                  <IconButton
                    tooltip="Description here"
                    hoveredStyle={classes.show}
                  >
                    <EditIcon onClick={showEdit}>Edit</EditIcon>
                  </IconButton>
                ) : (
                  <ImageUpload
                    imgData={data.profilePictrue}
                    changeUrl={(newUrl) => {
                      const getResponseDetails = localStorage.getItem(
                        'storeResponseDetails'
                      );
                      const getResponseDetailsAfterParse = JSON.parse(
                        getResponseDetails
                      );
                      updateProfilePic(getResponseDetailsAfterParse.id, newUrl);
                      this.setState({
                        data: {
                          ...this.state.data,
                          profilePictrue: newUrl.urlLink,
                        },
                        hidden: true,
                      });
                      console.log(
                        'Your profile pic that needs to be updated',
                        newUrl,
                        'data'
                      );
                    }}
                  />
                )}
              </Grid>
            </Grid>
          </div>
          <div>
            <br />
            <Grid container spacing={1} alignItems="flex-end">
              <Grid item>
                <AccountCircle />
              </Grid>

              <Grid item>
                <TextField
                  name="name"
                  type="text"
                  onChange={onModelChange}
                  placeholder="Name"
                  value={data.name}
                />
              </Grid>
            </Grid>
            {/* <Grid container spacing={1} alignItems="flex-end">
              <Grid item>
                <AccountCircle />
              </Grid>

              <Grid item>
                <TextField
                  name="password"
                  type="text"
                  hidden={true}
                  secureTextEntry={true}
                  onChange={onModelChange}
                  placeholder="password"
                  // value={data.password}
                />
            </Grid>
              </Grid> */}
          </div>

          {/* <div className={classes.margin}>
            <Grid container spacing={1} alignItems="flex-end">
              <Grid item>
                <LocalPhoneIcon />
              </Grid>
              <Grid item>
                <TextField
                  name="phone"
                  type="number"
                  // onChange={onModelchange}
                  placeholder="Phone"
                />
              </Grid>
            </Grid>
          </div> */}

          <div className>
            <Grid container spacing={1} alignItems="flex-end">
              <Grid item>
                <EmailIcon />
              </Grid>
              <Grid item>
                <TextField
                  id="input-with-icon-grid"
                  name="email"
                  type="text"
                  onChange={onModelChange}
                  placeholder="Email"
                  value={data.email}
                />
              </Grid>
            </Grid>
          </div>

          <br />
          <Button
            style={{ float: 'right', backgroundColor: 'skyblue' }}
            className={classes.button}
            onClick={sendDetails}
            variant="contained"
            size="small"
            startIcon={<SaveIcon />}
          >
            Save
          </Button>
        </TabPanel>
        <TabPanel value={value} spacing={2} index={0}></TabPanel>
        <TabPanel value={value} index={2}>
          <Typography>
            YOYO Balance
            <br /> {data.yoyobalance}
            <br />
            <br />
            <Button
              className={classes.button}
              style={{ float: 'right', backgroundColor: 'skyblue' }}
              variant="contained"
              size="small"
              onClick={this.openDialogBox}
            >
              Add Points
            </Button>
          </Typography>
          <DialogBox
            open={this.state.openDialog}
            closeDialogBox={this.closeDialogBox}
            updatePoints={this.updatePoints}
          ></DialogBox>
        </TabPanel>
        <TabPanel value={value} index={3}>
          {data.sentdHistory ? (
            <Typography>{data.sentdHistory}</Typography>
          ) : (
            <div>
              <Grid container spacing={3} alignItems="flex-end">
                <Grid item style={{ float: 'right' }}>
                  <Avatar
                    type=""
                    // src={data.profilePictrue}
                    src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcSkkolxSNKrb80PEcRLo2ocTotIoM1TmxA_U02TJrM0UGSdwP9H&usqp=CAU"
                    style={{ width: '90px', height: '100px' }}
                  />
                  <br />
                  <Typography className={classes.title} gutterBottom>
                    You have not sent any gift card! Please send .
                    <hr />
                    <br />
                    <Button
                      className={classes.button}
                      style={{ float: 'right', backgroundColor: 'skyblue' }}
                      variant="contained"
                      size="small"
                    >
                      click for send
                    </Button>
                  </Typography>
                </Grid>
              </Grid>
            </div>
          )}
        </TabPanel>
        <TabPanel value={value} index={4}>
          {data.receivedHistory ? (
            <Typography>{data.receivedHistory}</Typography>
          ) : (
            <Typography className={classes.title} color="primary" gutterBottom>
              You have not sent any gift card! Please send .
              <hr />
              <br />
            </Typography>
          )}
        </TabPanel>
      </div>
    );
  }
}
Profie.propTypes = {
  classes: PropTypes.object,
  root: PropTypes.object,
  margin: PropTypes.object,
  tab_design: PropTypes.string,
  newUrl: PropTypes.object,
};

export default withStyles(useStyles)(Profie);
