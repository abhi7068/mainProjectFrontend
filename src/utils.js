import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import reducer from '../src/store/reducers/rootReducer';
import { Router } from 'react-router-dom';

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const history = createMemoryHistory();
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  }
  Wrapper.propTypes = {
    children: PropTypes.elementType,
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
export * from '@testing-library/react';
export { render };
