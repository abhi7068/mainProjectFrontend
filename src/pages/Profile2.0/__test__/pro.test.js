import React from 'react';
import Profile from '../pro';
// import { render } from '../../../utils';
import { cleanup, fireEvent } from '@testing-library/react';
import { render } from '../../../utils';
import '@testing-library/jest-dom';
const singleUserDetail = {
  data: {
    cart: [],
    email: 'raje0096@gmail.com',
    isAdmin: false,
    name: 'Rajesh',
    receivedHistory: [],
    sentHistory: [],
    __v: 0,
    _id: '5e9e9435f978e3804cc48333',
  },
  status: 'SUCCEEDED',
};

afterEach(cleanup);
it('it render withour crashing', () => {
  const { asFragment } = render(<Profile />, {
    intialState: { singleUserDetail },
  });
  expect(
    asFragment(<Profile />, {
      intialState: { singleUserDetail },
    })
  ).toMatchSnapshot();
});

test('it render withour crashing', () => {
  const { getByTestId } = render(<Profile />, {
    intialState: { singleUserDetail },
  });
  expect(fireEvent.click(getByTestId('dialogopen'))).toBe(true);
  // expect(fireEvent.click(getByTestId('"show-data"'))).toBe(true);
});

// expect(getByTestId('show-data')).toHaveTextContent('Yoyobalance');
// expect(fireEvent.click(getByTestId('"show-data"'))).toBe(true);
