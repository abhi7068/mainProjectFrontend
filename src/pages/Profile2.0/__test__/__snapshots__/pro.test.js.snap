// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it render withour crashing 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div />
    <div
      class="MuiGrid-root pro-disp-2 MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-4 MuiGrid-grid-lg-4"
      >
        <img
          alt="Cindy Baker"
          class="image"
        />
        <button
          class="edit-icon"
          tabindex="0"
          tooltip="Description here"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-8 MuiGrid-grid-lg-8"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-true"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
            style="text-align: left;"
          >
            <div
              class="MuiPaper-root MuiCard-root pro-root-1 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                class="MuiCardHeader-root card"
                style="text-align: left;"
              >
                <div
                  class="MuiCardHeader-avatar"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="MuiCardHeader-content"
                />
              </div>
              <div
                class="MuiCardContent-root"
                style="text-align: left;"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-colorSecondary"
                />
                <ul
                  class="MuiList-root MuiList-padding"
                >
                  <li
                    class="MuiListItem-root MuiListItem-gutters"
                  >
                    <div
                      class="MuiListItemIcon-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M21 18v1c0 1.1-.9 2-2 2H5c-1.11 0-2-.9-2-2V5c0-1.1.89-2 2-2h14c1.1 0 2 .9 2 2v1h-9c-1.11 0-2 .9-2 2v8c0 1.1.89 2 2 2h9zm-9-2h10V8H12v8zm4-2.5c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiListItemText-root"
                      data-testid="show-data"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        Yoyobalance 
                      </span>
                    </div>
                  </li>
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                    role="button"
                    tabindex="0"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeSmall MuiButton-sizeSmall"
                      data-testid="dialogopen"
                      style="background-color: skyBlue;"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        ADD MORE
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                </ul>
                <p />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiContainer-root makeStyles-root-271 MuiContainer-maxWidthLg"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-8 MuiGrid-grid-lg-8"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-true"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
            >
              <div
                aria-label="icon label tabs example"
                class="MuiTabs-root"
              >
                <div
                  class="MuiTabs-scroller MuiTabs-fixed"
                  style="overflow: hidden;"
                >
                  <div
                    class="MuiTabs-flexContainer"
                    role="tablist"
                  >
                    <button
                      aria-selected="true"
                      class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary Mui-selected MuiTab-labelIcon"
                      role="tab"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiTab-wrapper"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2zm0 15l-5-2.18L7 18V5h10v13z"
                          />
                        </svg>
                        SendHistory
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      aria-selected="false"
                      class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary MuiTab-labelIcon"
                      role="tab"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiTab-wrapper"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M20 5.41L18.59 4 7 15.59V9H5v10h10v-2H8.41z"
                          />
                        </svg>
                        RecievedHistory
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <span
                    class="PrivateTabIndicator-root-296 PrivateTabIndicator-colorSecondary-298 MuiTabs-indicator"
                    style="left: 0px; width: 0px;"
                  />
                </div>
              </div>
              <div
                aria-labelledby="vertical-tab-0"
                class="MuiTypography-root tabPanel MuiTypography-body1"
                id="vertical-tabpanel-0"
                role="tabpanel"
              >
                <div
                  class="MuiBox-root MuiBox-root-301"
                >
                  <div
                    class="makeStyles-root-271"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-item MuiGrid-grid-xs-true"
                    >
                      <div
                        class="MuiGrid-root grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-lg-12"
                      >
                         
                        <div>
                          <p
                            class="MuiTypography-root MuiTypography-body1"
                          >
                            You have not sent any gift card! Please send .
                          </p>
                          <hr />
                          <p />
                        </div>
                      </div>
                    </div>
                  </div>
                   
                </div>
              </div>
              <div
                aria-labelledby="vertical-tab-1"
                class="MuiTypography-root tabPanel MuiTypography-body1"
                hidden=""
                id="vertical-tabpanel-1"
                role="tabpanel"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
